### ISSUE TITLE
Cannot export calendars - Exception thrown

### ISSUE THREAD

**Issue by CapSel** [nextcloud/calendar#4863]
### Steps to reproduce

1. Go to Calendar App
2. Click "pencil" next to calendar you want to export - for example "Personal"
3. Click export (nothing happens)
4. Click export (empty page)

### Expected behavior

I wish to have an ICS file that I could import into another instance of nextcloud.

### Actual behaviour

Nothing is shown.

### Calendar app version

4.2.0

### CalDAV-clients used

Web browser

### Browser

Newest firefox (108?)

### Client operating system

Arch Linux, MacOS

### Server operating system

ArchLinux, Ubuntu with dockerized nextcloud:fpm-alpine

### Web server

Nginx

### Database engine version

PostgreSQL

### PHP engine version

None

### Nextcloud version

25.0.2

### Updated from an older installed version or fresh install

Fresh install

### Log file

```Text
{"reqId":"XH5JPZQHzAEEskBLQ4SJ","level":3,"time":"2022-12-31T11:24:39+00:00","remoteAddr":"89.69.69.250","user":"capsel","app":"index","method":"POST","url":"/index.php/apps/calendar/v1/config/view","message":"OCA\\Calendar\\Controller\\SettingsController::setConfig(): Argument #2 ($value) must be of type string, null given, called in /var/www/html/lib/private/AppFramework/Http/Dispatcher.php on line 225 in file '/var/www/html/custom_apps/calendar/lib/Controller/SettingsController.php' line 71","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:108.0) Gecko/20100101 Firefox/108.0","version":"25.0.2.3","exception":{"Exception":"Exception","Message":"OCA\\Calendar\\Controller\\SettingsController::setConfig(): Argument #2 ($value) must be of type string, null given, called in /var/www/html/lib/private/AppFramework/Http/Dispatcher.php on line 225 in file '/var/www/html/custom_apps/calendar/lib/Controller/SettingsController.php' line 71","Code":0,"Trace":[{"file":"/var/www/html/lib/private/AppFramework/App.php","line":172,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Calendar\\Controller\\SettingsController"},"setConfig"]},{"file":"/var/www/html/lib/private/Route/Router.php","line":298,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Calendar\\Controller\\SettingsController","setConfig",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},["view","calendar.settings.setConfig"]]},{"file":"/var/www/html/lib/base.php","line":1047,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/apps/calendar/v1/config/view"]},{"file":"/var/www/html/index.php","line":36,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","Line":165,"Previous":{"Exception":"TypeError","Message":"OCA\\Calendar\\Controller\\SettingsController::setConfig(): Argument #2 ($value) must be of type string, null given, called in /var/www/html/lib/private/AppFramework/Http/Dispatcher.php on line 225","Code":0,"Trace":[{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":225,"function":"setConfig","class":"OCA\\Calendar\\Controller\\SettingsController","type":"->","args":["view",null]},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":133,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Calendar\\Controller\\SettingsController"},"setConfig"]},{"file":"/var/www/html/lib/private/AppFramework/App.php","line":172,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Calendar\\Controller\\SettingsController"},"setConfig"]},{"file":"/var/www/html/lib/private/Route/Router.php","line":298,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Calendar\\Controller\\SettingsController","setConfig",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},["view","calendar.settings.setConfig"]]},{"file":"/var/www/html/lib/base.php","line":1047,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/apps/calendar/v1/config/view"]},{"file":"/var/www/html/index.php","line":36,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/var/www/html/custom_apps/calendar/lib/Controller/SettingsController.php","Line":71},"CustomMessage":"--"},"id":"63b01c028a013"}
```

**Comment by CapSel** [comment]
I have this issue on:
- freshly installed Nextcloud in Docker from tag fpm-alpine + postgres:latest with nginx outside of container on ubuntu
- old instance (as in fresh install was looong time ago) of Nextcloud unpacked (not from distro package or docker) on ArchLinux with Apache and php-fpm

**Comment by miaulalala** [comment]
Please make sure to look for existing issues frst!

Duplicate of #4844

**Comment by tcitworld** [comment]
From what I can found, `setTimezone` and `setInitialView` don't check that they have indeed a valid value before doing the call to set the value.

**Comment by miaulalala** [comment]
Issue is also solved by #4873 : Vue-router (:to="") should only be used for internal/frontend routes and not for external routes. The initialization of route params goes crazy because of this and finally causes the backend issues.
